Stevenxl questions destroy

StevenXL opened the pull request:
Note: In the `destroy` action of the Questions controller, in the case in which a user cannot delete a question, they will be redirected back to that question. This breaks with our other routes in which we **render** a specific view. I did this because the show view depends on a lot of items, and re-creating that view here would lead to a lot of duplication. 

(I can always change this action to be consistent with our other actions if there is agreement / a good reason to do so).

StevenXL committed:
Add the destroy action to the Questions controller.

User can only destroy a question if s/he is the owner of that question.

StevenXL committed:
Refactor the layout file to use the content_tag vs. manually inserting
HTML.

StevenXL committed:
Add a 'Delete Question' button if the current user is the owner of the
current question.

StevenXL committed:
Add the 'question#destroy' route to the router.

StevenXL committed:
Add ability for user to delete their own question.

Moved mock of current_user method inside the '#create' block to make it
easier to test other controller actions.

dcr8898 commented on app/controllers/questions_controller.rb:
```diff
@@ -34,10 +34,22 @@ def unanswered
+      redirect_to question
+    end
+  end
+
```
This if/else should not be necessary.  I know it works, but the more common way to manage authorizations is using before_action methods to bar access to the action (destroy, in this case) completely.  You shouldn't have to manually check user_ids _in_ the action.

See the link I sent you for an example of requiring a logged in user.  The same method could be used to check the current_user.id against the record's owner ID and simply redirect requests away from the protected route.

StevenXL committed:
Updated the questions resource to indicate that we are using all routes.

StevenXL committed:
Add before actions to Questions controller.

Before action "logged_in_user" verifies that a user is logged in for all
actions except index, show, and unanswered.

Before action "verify_ownership" verifies that a user is the owner of
the question being modified for the update and destroy actions.

StevenXL committed:
Modify test suite for the Questions controller to not test the scenario
where the user is not authorized to modify / delete a question.

StevenXL committed:
Remove authrization from the Questions controller.

This will be implemented separately.
